src/sinks/datadog/metrics.rs

```diff
@@ -281,66 +218,106 @@ fn encode_events(events: Vec<Metric>, interval: i64, namespace: &str) -> Datadog
+                                metric,
+                                r#type,
+                                interval: Some(interval),
+                                points: vec![DatadogPoint(ts, value)],
+                                tags: tags.clone(),
+                            };
+                            match statistic {
+                                // https://docs.datadoghq.com/developers/metrics/metrics_type/?tab=histogram#metric-type-definition
+                                StatisticKind::Histogram => {
+                                    let mut result = vec![
+                                        metric(
+                                            format!("{}.min", &fullname),
+                                            DatadogMetricType::Gauge,
+                                            s.min,
+                                        ),
+                                        metric(
+                                            format!("{}.avg", &fullname),
+                                            DatadogMetricType::Gauge,
+                                            s.avg,
+                                        ),
+                                        metric(
+                                            format!("{}.count", &fullname),
+                                            DatadogMetricType::Rate,
+                                            s.count,
+                                        ),
+                                        metric(
+                                            format!("{}.median", &fullname),
+                                            DatadogMetricType::Gauge,
+                                            s.median,
+                                        ),
+                                        metric(
+                                            format!("{}.max", &fullname),
+                                            DatadogMetricType::Gauge,
+                                            s.max,
+                                        ),
+                                    ];
+
+                                    for (q, v) in s.quantiles {
+                                        result.push(metric(
+                                            format!(
+                                                "{}.{}percentile",
+                                                &fullname,
+                                                (q * 100.0) as u32
+                                            ),
+                                            DatadogMetricType::Gauge,
+                                            v,
+                                        ))
+                                    }
+
+                                    result
+                                }
+                                // https://docs.datadoghq.com/developers/metrics/types/?tab=distribution#definition
```

Comment:
If I'm reading these docs correctly, it sounds like we should not be aggregating anything at all here:

> Unlike the HISTOGRAM metric type, which aggregates on the Agent during a given time interval, a DISTRIBUTION metric sends all the raw data during a time interval to Datadog, and aggregations occur server-side.

I think the aggregations you've defined here are the ones DataDog will calculate on their end, not the ones they expect to receive.

Reply:
True, I'll change it.

Reply:
It can't be done over Datadog's HTTP API. It doesn't have a way to submit Distribution points. We need to specify `interval` for the points, but their documentation says it's to be defined for `rate` and `count` only. https://docs.datadoghq.com/api/v1/metrics/

We can document this behavior on the website, it shouldn't be too suprising. 

`statsd` sink can be used to get this behavior.

Reply:
Sorry for the slow response here @ktff. I've been digging into some DataDog code to see how these are sent and I think I have a pretty good idea.

The in-app libraries seem to forward these points to the datadog agent over the statsd protocol with the `d` type indicator. Within the agent, it seems that the individual points are aggregated into Datadog's custom [sketch](https://github.com/DataDog/sketches-go) data structure and then forwarded via an undocumented v2 API based on protocol buffers.

This is important because if we're not sending the individual points or that custom sketch type, we do not actually support DataDog distributions. That's way we're summarizing the points here does not allow them to be merged together on the server to determine quantiles across a whole infrastructure, which is the whole point of distributions over histograms.

As far as I can tell, there is no straightforward way for us to support this. We would have to implement DDSketch as a data structure internally and then reverse-engineer how to serialize and send them over the new undocumented API. We can't just fake it by aggregating locally since the whole point is to have the ability to aggregate globally.

Reply:
> We can't just fake it by aggregating locally since the whole point is to have the ability to aggregate globally.

True.

> there is no straightforward way for us to support this

I agree.